### DISCUSSION TITLE
method request: `isValid` method that returns whether the passed data is valid or not

### DISCUSSION THREAD

**Discussion by anuraglol** [colinhacks/zod#1951] [Q&A] [answered]
**About the issue:**

If you want to check whether the passed data is valid, you have to parse and bind a catch method to it.

In small use cases like checking whether a single input field passes the check, this could be quick and helpful, being a one-liner.

**Comment by JacobWeisenburger** [comment]
Are you able to give a code example?

**Comment by anuraglol** [comment]
> Are you able to give a code example?

Sure, so related to the above example, I had to write this piece of code

```ts
const isValidMail = useMemo(() => {
    try {
      emailSchema.parse(email);
      return true;
    } catch (error) {
      return false;
    }
  }, [email]);
```
  
This could easily be turned into a one-liner, making it suitable for small use cases where you just have to validate one state or so, instead of using a solution like `react-hook-form`, we could just do this:

```ts
const isValidMail = useMemo(() => {
    return z.isValid(emailSchema, email);
  }, [email]);
```

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

```ts
const isValidMail = useMemo(() => {
    return emailSchema.safeParse(email).success
}, [email])
```

**Comment by anuraglol** [comment]
> Is this what you are looking for?
> 
> ```ts
> const isValidMail = useMemo(() => {
>     return emailSchema.safeParse(email).success
> }, [email])
> ```

ah 
yes, thanks!
didn't know about that 😅
